### PR TITLE
[board] avaota-a1: set GICR_WAKER bit 2 for BL31

### PR DESCRIPTION
**Pull request by BigfootACA**
GICR_WAKER is GIC Redistributor Power Management Control Register, the second bit is Processor Sleep (WAKER_CA_BIT in BL31)

BL31 require GICR_WAKER bit 2 to be 1, otherwise it crashes.

`ASSERT: drivers/arm/gic/v3/gicv3_helpers.c:28:(gicr_read_waker(gicr_base) & WAKER_CA_BIT) != 0U`

when boot from SD Card, GICR_WAKER is 0x6
when boot from USB Fel, GICR_WAKER is 0x0

simply setting this bit to 1 will make bl31 to work properly

### CODE REVIEW

**Comment by YuzukiTsuru** [comment]
Move gicr_set_waker to jump to bl31, if change it in start.S you will not able back to USB FEL.

File: board/avaota-a1/start.S [hunk 1]
```diff
@@ -163,6 +163,8 @@ reset:
 
 	bl neon_enable
 
+	bl gicr_set_waker
```
Comment (YuzukiTsuru) [inline board/avaota-a1/start.S:166]: don't set gicr on this place, set it in the time need to get into bl31

File: board/avaota-a1/board.c [hunk 2]
```diff
@@ -164,6 +164,14 @@ void neon_enable(void) {
     asm volatile("MCR p10, 7, r3, c8, c0, 0");
 }
 
+void gicr_set_waker(void) {
+    uint32_t gicr_waker = read32(SUNXI_CPU_GIC600_BASE + 0x60014);
```
Comment (YuzukiTsuru) [inline board/avaota-a1/board.c:168]: do not use `0x60014`, add new define in reg-ncat.h

**Review by YuzukiTsuru — CHANGES_REQUESTED** [review]
do not use 0x60014, add new define in reg-ncat.h
like
```
#define GIC_IROUTR(_n)       (SUNXI_GIC600_BASE + 0x6000 + 8 * (_n))
#define GICR_LPI_BASE(n)     (SUNXI_GIC600_BASE + 0x60000 + n*0x20000)
#define GICR_WAKER(m)        (GICR_LPI_BASE(m) + 0x0014)
#define GICR_PWRR(m)         (GICR_LPI_BASE(m) + 0x0024)
#define LEVEL_TRIGERRED      (0)
#define EDGE_TRIGERRED       (1)
#define GIC_IRQ_TYPE_CFG(_n)	(SUNXI_GIC600_BASE + 0xc00 + 4 * (_n))
#define GIC_IRQ_MOD_CFG(_n)	(SUNXI_GIC600_BASE + 0xd00 + 4 * (_n))
```

using `gicr_set_waker` in  https://github.com/YuzukiHD/SyterKit/blob/58ebe1175459d3275e3541bc2f781662680dacb2/board/avaota-a1/extlinux_boot/main.c#L821-L825

```
    clean_syterkit_data();
    gicr_set_waker();
    jmp_to_arm64(CONFIG_BL31_LOAD_ADDR);
```

**Comment by BigfootACA** [comment]
all problems are fixed, thank!

**Review by YuzukiTsuru — APPROVED** [review]

**Pull request merged**